library/rmc-annotations/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// #![feature(register_tool)]
+// #![register_tool(rmctool)]
+// Frustratingly, it's not enough for our crate to enable these features, because we need all
+// downstream crates to enable these features as well.
+// So we have to enable this on the commandline (see rmc-rustc) with:
+//   RUSTFLAGS="-Zcrate-attr=feature(register_tool) -Zcrate-attr=register_tool(rmctool)"
+
+// proc_macro::quote is nightly-only, so we'll cobble things together instead
+use proc_macro::TokenStream;
+
+#[cfg(not(rmc))]
+#[proc_macro_attribute]
+pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    // Leave the code intact, so it can be easily be edited in an IDE,
+    // but outside RMC, this code is likely never called.
+    let mut result = TokenStream::new();
+
+    result.extend("#[allow(dead_code)]".parse::<TokenStream>().unwrap());
+    result.extend(item);
+    result
+    // quote!(
+    //     #[allow(dead_code)]
+    //     $item
+    // )
+}
+
+#[cfg(rmc)]
+#[proc_macro_attribute]
+pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let mut result = TokenStream::new();
+
+    result.extend("#[rmctool::proof]".parse::<TokenStream>().unwrap());
```

Comment:
I thought the need for `no_mangle` was just the UI problem of predicting the mangled name. This feature is intended to solve that, since we get a map:

```
{
      "pretty_name": "testing",
      "mangled_name": "_RNvCs1NZMV7qnTwy_1f7testing",
      "original_file": "/home/ubuntu/experiments/proof_anno/f.rs"
}
```

Is there some other reason for `no_mangle`? Otherwise, I don't know why I'd add it here.

Reply:
I was thinking about addressing that in a follow up PR so this PR doesn't get too big and we can merge it sooner rather than later. If you do add that to this PR, then it's OK to keep it without the no_mangle.

There are two reasons why we rely on the `#[no_mangle]` today. One is the friendly name and the second is that no_mangle makes the function public.